src/code/PreRender.c

```diff
@@ -1,42 +1,615 @@
-#include <ultra64.h>
 #include <global.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/PreRender/func_800C0E70.s")
+void PreRender_SetValuesSave(PreRenderContext* this, u32 width, u32 height, void* fbuf, void* zbuf, void* cvg)
+{
+    this->widthSave = width;
+    this->heightSave = height;
+    this->fbufSave = fbuf;
+    this->cvgSave = cvg;
+    this->zbufSave = zbuf;
+    this->ulxSave = 0;
+    this->ulySave = 0;
+    this->lrxSave = width-1;
+    this->lrySave = height-1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/PreRender/func_800C0EA8.s")
+void PreRender_Init(PreRenderContext* this)
+{
+    bzero(this, sizeof(PreRenderContext));
+    ListAlloc_Init(&this->alloc);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/PreRender/func_800C0ED8.s")
+void PreRender_SetValues(PreRenderContext* this, u32 width, u32 height, void* fbuf, void* zbuf)
+{
+    this->width = width;
+    this->height = height;
+    this->fbuf = fbuf;
+    this->zbuf = zbuf;
+    this->ulx = 0;
+    this->uly = 0;
+    this->lrx = width-1;
+    this->lry = height-1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/PreRender/func_800C0F08.s")
+void PreRender_Destroy(PreRenderContext* this)
+{
+    ListAlloc_FreeAll(&this->alloc);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/PreRender/func_800C0F28.s")
+void func_800C0F28(PreRenderContext* this, Gfx** gfxp, void* buf, void* bufSave)
+{
+    Gfx* gfx;
```

Comment:
do we want to go with the normal convention of calling gfx a dList and dListHead or something? not sure if this context is different and warrants a different name

Reply:
I've seen both being used. Not sure which name I should use, also maybe the OG name is good enough?

Reply:
I prefer dList if it actually is a display list, gfx doesnt tell me much. but thats just me

Reply:
Gfx can only be a display list anyway.